ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+# Orleans and Akka Actors: A Comparison
+
+## Step-By-Step Review
+
+This overview walks along the Technical Report [MSR-TR-2014-41](http://research.microsoft.com/apps/pubs/default.aspx?id=210931) by *Philip A. Bernstein, Sergey Bykov, Allan Geller, Cabriel Kliot, Jorgen Thelin (Microsoft Research)*, for another comparison focusing on Erlang and Riak see [Christopher Meiklejohn’s excellent article](http://christophermeiklejohn.com/orleans.html).
+
+### Introduction
+
+Different focus:
+
+  * Akka as a toolkit for building distributed systems, offering the full power but also exposing the inherent essential complexity.
+  
+  * Orleans restricts applicability in order to allow seamless use without understanding distributed computing.
```

Comment:
The intent of the point-by-point comparison is to highlight the differences between both approaches, not to sell one or the other. Therefore since both Orleans and Akka provide extensibility, the interesting question in this particular section is what each solution targets, and reading the TR I came to the conclusion that—bluntly speaking—Akka requires developers to understand distributed computing while Orleans aims at avoiding the necessity for that. This is supported by quotes such as

> To build a correct solution to such problems in the application, the developer must be a distributed systems expert. To avoid these complexities, we built the Orleans programming model and runtime, which raises the level of the actor abstraction.

and

> This level of indirection provides the runtime with the opportunity to solve many hard distributed systems problems that must otherwise be addressed by the developer.

Since I assume that you are one of the co-authors of the paper, you are in a good position to answer the question: what is the primary focus of Orleans? It is completely fine if Orleans also does other things as secondary concerns, but I get the feeling that the primary focus of Orleans and Akka is fundamentally different, so that is what I would like to highlight here.

I’ll flesh out the intro of “different focus” to clarify this intent of this section.